code/modules/mob/dead/new_player/new_player.dm

```diff
@@ -36,16 +36,23 @@
 	var/output = "<center><p><a href='byond://?src=[REF(src)];show_preferences=1'>Setup Character</a></p>"
 
 	if(SSticker.current_state <= GAME_STATE_PREGAME)
-		switch(ready)
-			if(PLAYER_NOT_READY)
-				output += "<p>\[ [LINKIFY_READY("Ready", PLAYER_READY_TO_PLAY)] | <b>Not Ready</b> | [LINKIFY_READY("Observe", PLAYER_READY_TO_OBSERVE)] \]</p>"
-			if(PLAYER_READY_TO_PLAY)
-				output += "<p>\[ <b>Ready</b> | [LINKIFY_READY("Not Ready", PLAYER_NOT_READY)] | [LINKIFY_READY("Observe", PLAYER_READY_TO_OBSERVE)] \]</p>"
-			if(PLAYER_READY_TO_OBSERVE)
-				output += "<p>\[ [LINKIFY_READY("Ready", PLAYER_READY_TO_PLAY)] | [LINKIFY_READY("Not Ready", PLAYER_NOT_READY)] | <b> Observe </b> \]</p>"
+		if(is_softbanned(src))
+			switch(ready)
+				if(PLAYER_READY_TO_OBSERVE)

```

Comment:
This code only controls if they are shown the button.

They could still spoof the ready up link or the join game button and the server would let them join.

PoC: 
1. join server
1. go to byond/cache in your documents
1. find the .html file for the ready window, it should be one of the first ones when sorted by modified 
    1.   (512 users, first find the subfolder for your DS instance inside byond cache, sorting by modified should work)
1. grab the ref for `src` from one of the `src=[0x....];ready=blah` links.
1. in the command bar, type `byond://?src=[ref from previous step];ready=1`
1. wait for game to start

Reply:
fixing rn (adding softban checks to Topic())